test/unit/context_unit_test.rb

```diff
@@ -466,4 +466,46 @@ def test_context_initialization_with_a_proc_in_environment
     assert contx
     assert_nil contx['poutine']
   end
+
+  def test_add_global_filter
```

Comment:
This is just a combination of what test_apply_global_filter and test_apply_global_filter_when_no_global_filter_exist does, so it can also be removed.